engine/src/main/java/com/redhat/cloud/notifications/processors/ExternalAuthorizationCriterion.java

```diff
@@ -3,34 +3,35 @@
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import io.vertx.core.json.JsonObject;
 import jakarta.validation.constraints.NotNull;
 import java.util.Objects;
 
 @JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
 @JsonIgnoreProperties(ignoreUnknown = true)
-public class ExternalAuthorizationCriteria {
+public class ExternalAuthorizationCriterion {
 
     @NotNull
-    private String assetType;
+    private JsonObject type;
```

Comment:
Why not have a dedicated object for the type? I think we will have to modify the `getLookupSubjectsRequest` to adjust the `.setName()` and the `.setNamespace()` calls.

Reply:
I was still dreaming to be agnostic of Kessel structure, but you're right, we have to be realistic :)